### PR TITLE
solution

### PR DESCRIPTION
**Pull request by maksym-kostetskyi**
https://maksym-kostetskyi.github.io/react_static-list-of-posts-js/

### CODE REVIEW

File: src/components/CommentInfo/CommentInfo.jsx [hunk 1]
```diff
@@ -1,3 +1,20 @@
-export const CommentInfo = () => (
-  <>Put the comment here</>
+export const CommentInfo = ({ comment }) => (
```
Comment (volodymyr-soltys97) [inline src/components/CommentInfo/CommentInfo.jsx:1]: Use a destruction for `comment`

File: src/components/CommentList/CommentList.jsx [hunk 2]
```diff
@@ -1,3 +1,10 @@
-export const CommentList = () => (
-  <>Put the list here</>
+import { CommentInfo } from "../CommentInfo";
+
+export const CommentList = ({ comments }) => (
+  <div className="CommentList">
+    {comments.map(comment => (
+      <CommentInfo comment={comment} />
```
Comment (volodymyr-soltys97) [inline src/components/CommentList/CommentList.jsx:6]: Where is the key attribute here?

File: src/components/PostInfo/PostInfo.jsx [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+
+export const PostInfo = ({ post }) => {
+  const comments = post.comments.length !== 0
+    ? <CommentList comments={post.comments} />
+    : <b data-cy="NoCommentsMessage">No comments yet</b>;
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{post.title}</h3>
```
Comment (volodymyr-soltys97) [inline src/components/PostInfo/PostInfo.jsx:11]: Use a destruction for `post`

File: src/components/PostList/PostList.jsx [hunk 4]
```diff
@@ -1,3 +1,9 @@
-export const PostList = () => (
-  <>Put the list here</>
+import { PostInfo } from "../PostInfo";
+
+export const PostList = ({ posts }) => (
+  <div className="PostList">
+    {posts.map(post => (
+      <PostInfo post={post} />
```
Comment (volodymyr-soltys97) [inline src/components/PostList/PostList.jsx:6]: The same as above, need to add the key attribute

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code

File: src/App.jsx [hunk 5]
```diff
@@ -1,107 +1,28 @@
...
+import usersFromServer from './api/users.json';
+
+function getUserById(userId) {
+  return usersFromServer.find(user => user.id === userId)
+      || null;
+}
+
+function getCommentsById(postId) {
+  return commentsFromServer.filter(comment => comment.postId === postId)
+      || null;
```
Comment (etojeDenys) [inline src/App.jsx:14]: null is redundant here, 'cause filter always returns the new array

File: src/components/PostInfo/PostInfo.jsx [hunk 6]
```diff
@@ -1,3 +1,28 @@
...
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{title}</h3>
+        <p>
+          {' Posted by  '}
+          <a className="UserInfo" href={`mailto:${user.email}`}>
+            {user.name}
+          </a>
```
Comment (etojeDenys) [inline src/components/PostInfo/PostInfo.jsx:17]: probably, you want to use the UserInfo component here, yeah?

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/components/PostInfo/PostInfo.jsx [hunk 7]
```diff
@@ -1,3 +1,27 @@
...
+    ? <CommentList comments={comments} />
+    : <b data-cy="NoCommentsMessage">No comments yet</b>;
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{title}</h3>
+        <p>
+          {' Posted by  '}
+          <UserInfo user={user} />
```
Comment (etojeDenys) [inline src/components/PostInfo/PostInfo.jsx:16]: looks good, but user can be null, you need to handle this case here

**Review by etojeDenys — CHANGES_REQUESTED** [review]

**Review by etojeDenys — APPROVED** [review]
good job

**Pull request closed without merging**